src/main/java/org/opensearch/knn/common/launchcontrol/KNNLaunchControl.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.common.launchcontrol;
+
+import org.opensearch.common.settings.Setting;
+import org.opensearch.knn.index.KNNSettings;
+
+import java.util.stream.Stream;
+
+public final class KNNLaunchControl {
+
+    // Feature flags
+    public static final String KNN_LAUNCH_QUERY_REWRITE_ENABLED = "knn.launch.query.rewrite.enabled";
+    private static final boolean KNN_LAUNCH_QUERY_REWRITE_ENABLED_DEFAULT = true;
+
+    public static Setting<Boolean> KNN_LAUNCH_QUERY_REWRITE_ENABLED_SETTING = Setting.boolSetting(
+        KNN_LAUNCH_QUERY_REWRITE_ENABLED,
+        KNN_LAUNCH_QUERY_REWRITE_ENABLED_DEFAULT,
+        Setting.Property.NodeScope,
+        Setting.Property.Dynamic
+    );
```

Comment:
Looking at the usage of this, this is not a true feature flag. because if we want to make this as feature flag we would have register it with KNNplugin class. 

But if we want to treat this as a setting then lets move this in settings.

Reply:
I wanted to distinguish between a setting and launch control flag. I used FeatureFlag before but the process of flipping it required a new distribution build and adding a new flag as a setting required a change in core. Both aren't ideal so chose plugin related settings route. 

A separate class makes the intent clear.

Reply:
When you say launch control? what launch it is controlling? 

> I used FeatureFlag before but the process of flipping it required a new distribution build 

Why? can you share some reference where you read it?

> adding a new flag as a setting required a change in core

Please share reference.

Reply:
> When you say launch control? what launch it is controlling?

NativeEngineQuery. I am changing it back to feature flag if thats the preference

> Why? can you share some reference where you read it?

https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/util/FeatureFlags.java#L169

> Please share reference.

https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/util/FeatureFlags.java#L138

Reply:
I double checked this and it doesn't work so have to go with the settings approach.